imap/sync_support.c

```diff
@@ -8048,9 +8057,31 @@ EXPORTED int sync_connect(struct sync_client_state *sync_cs)
     }
 #endif
 
+    unsigned capabilities = 0;
+
     if (CAPA(backend, CAPA_SIEVE_MAILBOX)) {
         syslog(LOG_INFO, "Destination supports #sieve mailbox");
-        sync_do_enable(sync_cs, CAPA_SIEVE_MAILBOX);
+        capabilities |= CAPA_SIEVE_MAILBOX;
+    }
+
+    if (sync_cs->flags & SYNC_FLAG_ARCHIVE) {
+        if (CAPA(backend, CAPA_REPLICATION_ARCHIVE)) {
+            syslog(LOG_INFO, "Destination supports replication to archive");
+            capabilities |= CAPA_REPLICATION_ARCHIVE;
+            sync_cs->flags &= ~SYNC_FLAG_ARCHIVE;
+        }
+        else {
+            syslog(LOG_NOTICE, "Destination doesn't support replication to archive");
+        }
+    }
+
+    if (capabilities) {
+        sync_do_enable(sync_cs, capabilities);
+
+        if (capabilities & CAPA_REPLICATION_ARCHIVE && !(sync_cs->flags & SYNC_FLAG_ARCHIVE)) {
```

Comment:
For clarity, I'd like to see this as:

`if ((capabilities & CAPA_REPLICATION_ARCHIVE) && !(sync_cs->flags & SYNC_FLAG_ARCHIVE)) {`

Reply:
Oh well spotted.  Usually my compiler complains about bitwise-and without parens in a comparison, I wonder why it didn't complain about this one?

Anyway: fixed it, and wrapped the long line too.